python/pyspark/pandas/tests/test_dataframe.py

```diff
@@ -270,7 +270,12 @@ def test_inplace(self):
         psdf["a"] = psdf["a"] + 10
 
         self.assert_eq(psdf, pdf)
-        self.assert_eq(psser, pser)
+        # SPARK-38946: Since Spark 3.4, inplace set generate a new dataframe to follow
+        # pandas 1.4 behaviors
+        if LooseVersion(pd.__version__) >= LooseVersion("1.4"):
+            self.assert_eq(psser, pser)
```

Comment:
Seems like this is a common test with the old pandas?
Shall we move it out of `if LooseVersion(pd.__version__) >= LooseVersion("1.4"):`?